viewer/packages/api/src/public/migration/Cognite3DViewer.ts

```diff
@@ -1631,6 +1682,42 @@ export class Cognite3DViewer {
   ): Promise<CameraManagerCallbackData> {
     const intersection = await this.intersectModels(offsetX, offsetY, { asyncCADIntersection: false });
 
+    const normalizedCoords = getNormalizedPixelCoordinates(this.domElement, offsetX, offsetY);
+
+    const result: CameraManagerCallbackData = {
+      intersection,
+      modelsBoundingBox: this._updateNearAndFarPlaneBuffers.combinedBbox,
+      pickedBoundingBox: undefined
+    };
+
+    let closestDistanceToCamera = result.intersection?.distanceToCamera ?? undefined;
+    let customObjectIntersectInput: CustomObjectIntersectInput | undefined = undefined; // Lazy creation for speed
+
+    this._sceneHandler.customObjects.forEach(customObject => {
+      if (!customObject.shouldPick) {
+        return;
+      }
+      if (!customObjectIntersectInput) {
+        customObjectIntersectInput = new CustomObjectIntersectInput(
+          normalizedCoords,
+          this.cameraManager.getCamera(),
+          this.getGlobalClippingPlanes()
+        );
+      }
+      const customObjectIntersection = customObject.intersectIfCloser(
+        customObjectIntersectInput,
+        closestDistanceToCamera
+      );
+      if (!customObjectIntersection) {
+        return;
+      }
+      closestDistanceToCamera = customObjectIntersection.distanceToCamera;
+      result.intersection = customObjectIntersection;
+      if (pickBoundingBox) {
+        result.pickedBoundingBox = customObjectIntersection.boundingBox;
+      }
+    });
```

Comment:
If I understand this correctly, we iterate through each custom object and add any applicable data from the intersections to the final result? Wouldn't it make more sense that all the data comes from just one intersection? So that we only return data from the first intersection from the ray, whether that's CAD, pointcloud or a custom object?